processing/src/main/java/org/apache/druid/query/filter/SpatialDimFilter.java

```diff
@@ -20,35 +20,49 @@
 package org.apache.druid.query.filter;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.RangeSet;
-import com.google.common.collect.Sets;
 import org.apache.druid.collections.spatial.search.Bound;
 import org.apache.druid.java.util.common.StringUtils;
 import org.apache.druid.segment.filter.SpatialFilter;
 
+import javax.annotation.Nullable;
 import java.nio.ByteBuffer;
-import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
 
 /**
  */
 public class SpatialDimFilter implements DimFilter
 {
   private final String dimension;
   private final Bound bound;
+  @Nullable
+  private final FilterTuning filterTuning;
```

Comment:
I totally agree here. Doing it externally seemed difficult because I wanted it to be per filter, but I did heavily consider making some sort of abstract class on top of `DimFilter` to handle `extractionFn` and `filterTuning`, but I think at this point would prefer to do as a follow-up so I will open an issue soon about refactoring `DimFilter`, and potentially merging `Filter` and `DimFilter` as well since there isn't a good reason for them to be split.

Reply:
created #8256